### ISSUE TITLE
Status Notifier support for tray icon

### ISSUE THREAD

**Issue by Solarunit** [Taiko2k/TauonMusicBox#1094]
Some time ago Budgie DE adopted Status Notifier system for tray icons:

>The System Tray applet was completely overhauled in Budgie 10.8. Historically we have used either na-tray or the XEmbed Protocol for the creation, management, and rendering of trays icons. In Budgie 10.8, our System Tray adopted the [Status Notifier](https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/) specification. This is a modern, vendor-agnostic solution that enables applications to communicate all relevant information for the rendering of their tray icon over DBus rather than being responsible for drawing tray icons and menus themselves, promoting the development of catered solutions by desktop environments and those developing panels for user experiences such as tiling window managers.

https://buddiesofbudgie.org/blog/state-of-the-budgie-2023

Because of this Tauon tray icon is no longer visible in the tray and :gear: icon is showed instead.

Could you add support of Status Notifier specification?

**Comment by Taiko2k** [comment]
I believe this should work as Tauon uses libappindicator which uses StatusNotifier. When you say a gear appears is that Tauons trap applet just without the icon? Or is it not appearing at all?

**Comment by Solarunit** [comment]
I can see icon, but it gear icon instead. Right click context menu works.
Maybe it's because I have flatpak version?

**Comment by Taiko2k** [comment]
Afaik it works in Gnome and KDE, so it's something about Budgies implementation that is different.

**Comment by Solarunit** [comment]
I am not sure, I asked Budgie devs about it. And it's not only tauon, some other apps too.

**Comment by Solarunit** [comment]
maybe it can't find the tauon (flatpak) icon path location

**Comment by Solarunit** [comment]
Looks like Status Notifier doesn't allow a full path to the icon.

**Comment by Taiko2k** [comment]
That's probably it. Maybe gnome and KDE break that spec, cause I'm pretty sure this works on wayland for those. 

So either sending the icon over dbus, or installing named icons? could be a solution here.

**Comment by Solarunit** [comment]
>sending the icon over dbus

does this mean sending the image data?

I was told either of those should probably work

**Comment by Taiko2k** [comment]
Yes I assume sending the pixel data is the propper way, what Tauon is doing with fullpaths to the user data folder is hack.

**Comment by Solarunit** [comment]
So could you add this "sending the icon over dbus" method?

**Comment by Taiko2k** [comment]
I'll add it to the Todo list, though anyone else is welcome to attempt a PR.

**Comment by Solarunit** [comment]
thanks

**Comment by Solarunit** [comment]
>I created issue for Budgie, so maybe we will see some workaround on Budgie side..

Update from Budgie side:

>it'll still be broken in Flatpak because the full path to icon can't be resolved outside the sandbox

**Comment by C0rn3j** [comment]
This could possibly be resolved by implementing #1316

**Comment by C0rn3j** [comment]
Closing in favor of that issue, #1772 should fix this when it can be implemented after SDL adds support for it.
